documentation/tutorial.md

```diff
@@ -0,0 +1,65 @@
+# Git Town Tutorial
+
+Let's assume you are part of a development team with Bob and Carol. Your repository is hosted on GitHub, and you follow [GitHub Flow](https://guides.github.com/introduction/flow/index.html).
+
+
+## Starting a new feature
+
+You are in the middle of the sprint, and have just finished a feature.
+You take the next ticket from the backlog.
+Let's say it is called "resetting passwords".
+You run `git hack reset-passwords`. This command
+
+* checks out the _master_ branch
+* pulls updates for the _master_ branch, i.e. the things that Bob and Carol have shipped while you worked on your last feature
+* cuts a new feature branch called "password-reset" from your now up-to-date _master_ branch
+* checks out that new branch
+
+This gives you the best possible start for the new password reset feature.
+
+
+## Synchronizing the branch
+
+After coding for a while, you hear that Bob shipped a number of important bug fixes,
+and that Carol shipped some UI design updates.
+Both affect your work, so you want them in your branch before you continue working.
+In order to achieve that, you run `git sync`. This command
+
+* stashes away your currently open changes
+* checks out the _master_ branch
+* pulls the updates on that branch (Bob's bug fixes and Carol's UI updates)
+* checks out your feature branch again
+* merges the updates from the _master_ branch into your branch
+* pushes your updated feature branch to the repository
+* restores your open changes by popping the stash
```

Comment:
Overall I think this tutorial is going about it in the wrong way. You describe a situation where you would use a git town command and then just dive into what the git town command does. Instead I think we need to describe the situation, describe what needs to be done, and then boom, introduce the command that does just what you need. Here is what I mean:

------------------------

After coding for a while, you hear that Bob shipped a number of important bug fixes,
and that Carol shipped some UI design updates. Both affect your work, so you want them in your branch before you continue working. On a high level you need to:

* pulls the updates for the __master__ branch (Bob's bug fixes and Carol's UI updates)
* merge the __master__ branch into your branch

You will need to move between branches to do this, which means you may need to stash or commit any open changes.

`git sync` encapsulates this process and brings you back to exactly where you were.